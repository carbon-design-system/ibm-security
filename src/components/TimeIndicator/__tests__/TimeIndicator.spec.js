/**
 * @file Time indicator tests.
 * @copyright IBM Security 2020
 */

import { render } from '@testing-library/react';
import React from 'react';

import { TimeIndicator } from '../../..';

describe('TimeIndicator', () => {
  test('should have no Axe or DAP violations', async () => {
    const main = document.createElement('main');
    render(<TimeIndicator>test content</TimeIndicator>, {
      // DAP requires a landmark '<main>' in the DOM:
      container: document.body.appendChild(main),
    });
    await expect(document.body).toHaveNoAxeViolations();
    await expect(document.body).toHaveNoDAPViolations('TimeIndicator');
  });

  test('should add custom class', () => {
    const { getByText } = render(
      <TimeIndicator className="custom-class">test content</TimeIndicator>
    );
    expect(getByText(/test content/i)).toHaveClass('custom-class');
  });

  test('should pass extra props through spread attribute', () => {
    const { queryByTestId } = render(<TimeIndicator data-testid="test-id" />);
    expect(queryByTestId('test-id')).toBeInTheDocument();
  });
});
