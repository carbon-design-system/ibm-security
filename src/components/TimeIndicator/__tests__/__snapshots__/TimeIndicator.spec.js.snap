// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeIndicator renders 1`] = `
<span
  class="security--time-indicator"
>
  <svg
    aria-hidden="true"
    class="security--time-indicator__icon"
    focusable="false"
    height="16"
    preserveAspectRatio="xMidYMid meet"
    style="will-change: transform;"
    viewBox="0 0 32 32"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16,30A14,14,0,1,1,30,16,14,14,0,0,1,16,30ZM16,4A12,12,0,1,0,28,16,12,12,0,0,0,16,4Z"
    />
    <path
      d="M20.59 22L15 16.41 15 7 17 7 17 15.58 22 20.59 20.59 22z"
    />
  </svg>
  10 minute setup
</span>
`;
